src/service/gif.js

```diff
@@ -1,9 +1,9 @@
 import api from './config';
 
-export const getGifFeatured = async () =>
+export const getGifFeatured = async (search) =>
   new Promise((resolve, reject) => {
     api
-      .get('/chat/gif/featured')
+      .get(search?.trim() !== '' ? `/chat/gif/search?q=${search.trim()}` : '/chat/gif/featured')
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new code introduces a potential security issue. The `search` parameter is directly interpolated into the URL without any sanitization or encoding, which could lead to an injection attack if the user input is not properly controlled. 

To fix this, you should use the `encodeURIComponent` function to encode the `search` parameter before interpolating it into the URL.

```diff
- .get(search?.trim() !== '' ? `/chat/gif/search?q=${search.trim()}` : '/chat/gif/featured')
+ .get(search?.trim() !== '' ? `/chat/gif/search?q=${encodeURIComponent(search.trim())}` : '/chat/gif/featured')
```
This will ensure that any special characters in the `search` string are properly encoded and cannot be used to manipulate the URL.


<!-- This is an auto-generated comment by OSS CodeRabbit -->

Reply:
Fixed